azure/deploymentframeworks/arm/output.ftl

```diff
@@ -128,6 +172,117 @@
     /]
 [/#macro]
 
+[#function getOutputMappingDataType type]
+    [#switch type]
+        [#case DICTIONARY_ATTRIBUTE_TYPE]
+            [#return "object"]
+            [#break]
+
+        [#default]
+            [#return "string"]
+            [#break]
+    [/#switch]
+[/#function]
+
+[#-- Deconstruct a resourceId value into discovered scopes --]
+[#function getResourceScopeFromResourcePath id]
+    [#local segments = getAzureResourcePropertySegments(id)]
+    [#local subscriptionIndex = segments?seq_index_of("subscriptions")!""]
+    [#local resourceGroupIndex = segments?seq_index_of("resourceGroups")!""]
+    [#local providerIndex = segments?seq_index_of("providers")!""]
```

Comment:
Does it matter that this will default to a different type?

Reply:
The defaults are there so as to not give an unhelpful freemarker error message if its missing, I'm not really sure how to default to an integer nicely (i.e without setting it to some arbitrary number). I've added some validation in 9146a1f - the defaulting to an empty string will get us past the thrown freemarker error, and then the validation will allow the calls to `@fatal` to resolve and display to the end user if necessary.